src/plugins/condition/ConditionSetTelemetryProvider.js

```diff
@@ -21,9 +43,16 @@ export default class ConditionSetTelemetryProvider {
         let conditionManager = new ConditionManager(domainObject, this.openmct);
         conditionManager.on('conditionSetResultUpdated', callback);
 
+        let stopObservingForChanges = this.openmct.objects.observe(domainObject, '*', (newDomainObject) => {
```

Comment:
This should be handled in the conditionManager. ConditionManager should receive the domainObject and listen for changes.

Reply:
@shefalijoshi , @deeptailor . #2748 fixes this. Please review and merge (into this branch) and then we can then also merge this PR.

Reply:
expensive updates to be fixed in #2750